force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -102,16 +119,18 @@ public with sharing class ClassesObjectsMethods {
      */
     public static Integer countVowels(String str) {
         // Initialize the result Integer
-
-        /*
-        for () {
+        Integer vowels = 0;
+        while(str.length() > 0) {
             // get individual characters from the string - hint use substring or split
+            String letter = str.substring(0, 1);
+            str = str.right(str.length()-1);
             // check if the character is a vowel
-            // if it is a vowel, increment the result Integer
+            if(letter == 'a' || letter == 'e' || letter == 'i' || letter == 'o' || letter == 'u'){
+           // if it is a vowel, increment the result Integer
+                vowels += 1;
+            }
         }
-        */
-
-        return null; // Replace null with the variable you used to store the result
+        return vowels; // Replace null with the variable you used to store the result
```

Comment:
Good practice here with String manipulation. My suggestions are to use a Set any time you are checking for membership and String.split() is probably the easiest way to break the string up into letters. 

```
 public static Integer countVowels(String str) {
        Integer vowelCount = 0;
        Set<String> vowels = new Set<String>{
                'a', 'e', 'i', 'o', 'u'
        };
        for (String character : str.split(''))  {
            if (vowels.contains(character.toLowerCase())) {
                vowelCount++;
            }
        }
        return vowelCount;
    }
 ```